float32 velocity
float32 distance
float32 angle
uint32 new_mesg
uint8 blocking